README.md

```diff
@@ -0,0 +1,93 @@
+# Find One
+
+The `find_one()` family of functions are wrappers for common WordPress retrieval functions like `get_posts()` or `get_terms()` that reduce the return value of those retrieval functions into a single result.
+
+For example, to find one item in a `person` post type:
+
+```php
+<?php
+
+use function Alley\WP\find_one_post;
+
+// Returns a \WP_Post or null.
+$person = find_one_post( 
+    [
+        'meta_key'   => 'twitter',
+        'meta_value' => '@potatomaster',
+        'post_type'  => 'person',
+    ] 
+);
+```
+
+which is equivalent to:
+
+```php
+<?php
+
+$person = \get_posts( 
+    [
+        'meta_key'         => 'twitter',
+        'meta_value'       => '@potatomaster',
+        'post_type'        => 'person',
+        'posts_per_page'   => 1,
+        'suppress_filters' => false,
+    ] 
+);
+
+if ( ! empty( $person[0] ) ) {
+	$person = $person[0];
+}
+```
+
+Taxonomy terms can be searched for similarly:
+
+```php
+<?php
+
+use function Alley\WP\find_one_term;
+
+// Returns a \WP_Term or null.
+$category = find_one_term( 
+    [
+        'slug'     => 'potatomaster',
+        'taxonomy' => 'category',
+    ] 
+);
+```
+
+which is equivalent to:
+
+```php
+<?php
+
+$category = \get_terms( 
+    [
+        'number'   => 1,
+        'slug'     => 'potatomaster',
+        'taxonomy' => 'category',
+    ] 
+);
+
+if ( ! empty( $category[0] ) ) {
+	$category = $category[0];
+}
+```
+
+The underlying `find_result()` function accepts a `$class` of object to search for and the array of query results:
+
+```php
+<?php
+
+use function Alley\WP\find_result;
+
+// Returns a \WP_Network or null.
+$network = find_result( \WP_Network::class, \get_networks() );
+```
+
+Helper functions are available for all core WordPress data types that have meta tables:
+
+* `find_one_comment()`
+* `find_one_post()`
+* `find_one_site()`
+* `find_one_term()`
+* `find_one_user()`
```

Comment:
Sorry, can you elaborate? What would the label look like? Do you have an example in mind of the documentation that would be linked to?

Reply:
```markdown
* **`find_one_comment()`** → [`WP_Comment`](https://developer.wordpress.org/reference/classes/wp_comment/)
* **`find_one_post()`** → [`WP_Post`](https://developer.wordpress.org/reference/classes/wp_post/)
* **`find_one_site()`** → [`WP_Site`](https://developer.wordpress.org/reference/classes/wp_site/)
* **`find_one_term()`** → [`WP_Term`](https://developer.wordpress.org/reference/classes/wp_term/)
* **`find_one_user()`** → [`WP_User`](https://developer.wordpress.org/reference/classes/wp_user/)
```

Rendered as:

* **`find_one_comment()`** → [`WP_Comment`](https://developer.wordpress.org/reference/classes/wp_comment/)
* **`find_one_post()`** → [`WP_Post`](https://developer.wordpress.org/reference/classes/wp_post/)
* **`find_one_site()`** → [`WP_Site`](https://developer.wordpress.org/reference/classes/wp_site/)
* **`find_one_term()`** → [`WP_Term`](https://developer.wordpress.org/reference/classes/wp_term/)
* **`find_one_user()`** → [`WP_User`](https://developer.wordpress.org/reference/classes/wp_user/)

Reply:
Since the functions can return non-objects, like a single post ID when using `'fields' => 'IDs'`, linking to the classes seems a little misleading, but let me ponder it.